include/revng/ADT/GenericGraph.h

```diff
@@ -77,22 +87,62 @@ struct std::tuple_element<1, Edge<Node, EdgeLabel>> {
 
 } // namespace std
 
+template<typename Node,
+         typename EdgeLabel,
+         bool HasParent,
+         size_t SmallSize,
+         template<typename, typename, bool, size_t, typename>
+         class ForwardEdge,
+         typename MostDerivedType>
+struct ForwardNodeBaseTCalc {
+  static constexpr bool
+    NoMostDer = std::is_same_v<MostDerivedType, std::false_type>;
```

Comment:
Please do not use abbreviations, they're not readable.

Reply:
i had to or clang tidy would have broken the line

Reply:
I suggest `IsFinal` and then use `!IsFinal`

Reply:
renamed in NoDerivation